### PR TITLE
Using lambda and CompleteFuture to decrease logic codes (#)

### PR DESCRIPTION
**Pull request by Technoboy-**

### CODE REVIEW

File: shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java [hunk 1]
```diff
@@ -139,15 +138,8 @@ private void doHealthCheck() {
...
-            String key = entry.getKey();
-            List<DivideUpstream> value = entry.getValue();
-            for (DivideUpstream upstream : value) {
-                CompletableFuture<UpstreamWithSelectorId> future = CompletableFuture.supplyAsync(() -> check(key, upstream), executor);
-
-                futures.add(future);
-            }
-        }
+        map.forEach((k, v) -> v.forEach(i -> futures.add(
+                CompletableFuture.supplyAsync(() -> check(k, i), executor).thenAccept(d -> putEntityToMap(d)))));
```
Comment (dengliming) [inline shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java:142]: `.thenAccept(d -> putEntityToMap(d)` That's not the original logic.

Comment (Technoboy-) [reply shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java:142]: The original logic is done in waitFinish.   That's not the correct way to use CompletableFuture.

Comment (Technoboy-) [reply shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java:142]: The original way to do putEntityToMap is in waitFinish, it's after `check` right ?

Comment (dengliming) [reply shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java:142]: Can you help to remove `waitFinish` method and use the correct way?

Comment (Technoboy-) [reply shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java:142]: Why remove `waitFinish` ? it's waiting for the futures to done.

Comment (dengliming) [reply shenyu-common/src/main/java/org/apache/shenyu/common/healthcheck/HealthCheckTask.java:142]: My mistake. You already removed ` putEntityToMap(entity); `from `waitFinish `

**Review by dengliming — APPROVED** [review]

**Pull request merged**